### DISCUSSION TITLE
Leaderboards for exercise tasks

### DISCUSSION THREAD

**Discussion by Jask999** [ls1intum/Artemis#9702] [Ideas]
It could be a cool idea to create leaderboards for exercise tasks. For example, a leaderboard listing who completed the task the quickest, or with the least commits, or most efficiently. Such a leaderboard would introduce a really fun and competitive aspect to Artemis; many people would enjoy it.

**Comment by MaximilianAnzinger** [comment]
Hi @Jask999 ,

this sounds like a great idea. Also I have thought about it. Although this is something I personally would enjoy, this might not be a suitable feature for all learners. While some might see this as a motivational tool, others can feel discouraged by it.

**Reply by bruderjakob17** [reply]
An idea would be to make it opt-in, and allowing students to choose a pseduonym.
